### PR TITLE
Ea 1693 registration with captcha

### PR DESCRIPTION
**Pull request by helinek**

### CODE REVIEW

File: apikey-service/pom.xml [hunk 1]
```diff
@@ -71,6 +71,12 @@
             <version>${spring-boot.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <version>20140107</version>
```
Comment (P-Ehlert) [inline apikey-service/pom.xml]: Why use such an old version and not the latest: https://mvnrepository.com/artifact/org.json/json/20190722?

File: apikey-service/src/main/java/eu/europeana/apikey/captcha/CaptchaManager.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    /**
+     * Post token in the verification request to the verification URL. Return JSON response or null in case of any exception.
+     *
+     * @param captchaToken token to be verified
+     * @return JSON response from the verification URL or null in case of any exception
+     */
+    private String getVerificationResponse(String captchaToken) {
+        try {
+            HttpPost httpPost = new HttpPost(getVerificationURI(captchaToken));
+            CloseableHttpResponse response = httpClient.execute(httpPost);
```
Comment (P-Ehlert) [inline apikey-service/src/main/java/eu/europeana/apikey/captcha/CaptchaManager.java]: response is not closed anywhere

File: apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java [hunk 3]
```diff
@@ -62,6 +65,21 @@
...
+    private String managerClientId;
+
+    @Value("${keycloak.manager-client-secret}")
+    private String managerClientSecret;
+
+    @Autowired
+    private CaptchaManager captchaManager;
+
+    @Autowired
+    private CustomKeycloakAuthenticationProvider customKeycloakAuthenticationProvider;
```
Comment (P-Ehlert) [inline apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java]: It's not a big issue, but we try not to use field injection and instead use constructor-based (see also https://www.vojtechruzicka.com/field-dependency-injection-considered-harmful/)

File: apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java [hunk 4]
```diff
@@ -143,6 +244,31 @@ public ApikeyController(ApikeyRepo apikeyRepo) {
...
 
+    /**
+     * Get value from the Authorization header of the given request based on the supplied pattern.
+     *
+     * @param httpServletRequest request with the header
+     * @param valuePattern pattern of the Authorization header to retrieve the value
+     * @return value of the Authorization header
+     */
+    private String getAuthorizationHeader(HttpServletRequest httpServletRequest, String valuePattern) {
+        String authorization = httpServletRequest.getHeader("Authorization");
```
Comment (P-Ehlert) [inline apikey-service/src/main/java/eu/europeana/apikey/controller/ApikeyController.java]: Very minor issue, but `getHeader("Authorization")` can be replaced with `getHeader(HttpHeaders.AUTHORIZATION)`

**Comment by helinek** [comment]
I implemented all the remarks.

**Pull request merged**